llvm/include/llvm/ADT/AddressRanges.h

```diff
@@ -164,15 +167,15 @@ inline bool operator==(const AddressRangeValuePair &LHS,
 /// ranges are not combined during insertion.
 class AddressRangesMap : public AddressRangesBase<AddressRangeValuePair> {
 public:
-  void insert(AddressRange Range, int64_t Value) {
+  void insert(AddressRange Range, const int64_t Value) {
     if (Range.empty())
       return;
 
     // Search for range which is less than or equal incoming Range.
-    auto It = std::partition_point(Ranges.begin(), Ranges.end(),
-                                   [=](const AddressRangeValuePair &R) {
-                                     return R.Range.start() <= Range.start();
-                                   });
+    auto *It = std::partition_point(Ranges.begin(), Ranges.end(),
```

Comment:
`Ranges` is a vector -- this does not have to return a pointer if we change the iterator representation